### PR TITLE
HAL added for absdiff(array, scalar) + related fixes

### PR DESCRIPTION
**Pull request by savuor**
### This PR changes
* HAL for `absdiff` when one of arguments is a scalar, including multichannel arrays and scalars
* several channels support for HAL `addScalar`
* proper data type check for `addScalar` when one of arguments is a scalar

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
```
 /home/ci/opencv/modules/core/src/arithm.cpp:964:22: warning: 'sdepth' may be used uninitialized in this function [-Wmaybe-uninitialized]

/home/ci/opencv/modules/core/src/arithm.cpp:1094:31: warning: 'sdepth' may be used uninitialized in this function [-Wmaybe-uninitialized]
```

File: modules/core/src/hal_replacement.hpp [hunk 1]
```diff
@@ -165,6 +166,23 @@ inline int hal_ni_absdiff16s(const short *src1_data, size_t src1_step, const sho
...
+@param src_step source image step
+@param dst_data destination image data
+@param dst_step destination image step
+@param width width of the images
+@param height height of the images
+@param scalar_data pointer to scalar value
+@param nChannels number of channels per element
+*/
+inline int hal_ni_absDiffScalar32f32f(const float* src_data, size_t src_step, float* dst_data, size_t dst_step, int width, int height, const float* scalar_data, int nChannels) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
+inline int hal_ni_absDiffScalar32s32s(const int*   src_data, size_t src_step, int*   dst_data, size_t dst_step, int width, int height, const int*   scalar_data, int nChannels) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```
Comment (vpisarev) [inline modules/core/src/hal_replacement.hpp:183]: shall it be unsigned/uint32_t type for the output array?

Comment (savuor) [reply modules/core/src/hal_replacement.hpp:183]: discussed, fixed

**Comment by savuor** [comment]
@asmorkalov I've changed output type from 32s to 32u:
* the actual value range of absdiff(32s, 32s) is 32u
* will be useful for future support in 5.x

**Review by vpisarev — APPROVED** [review]

**Pull request merged**